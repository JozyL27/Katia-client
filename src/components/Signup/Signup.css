.Signup__container {
  width: 70%;
  font-size: 1.6rem;
  /* border: .1rem solid black; */
  text-align: center;
  padding: 1rem 1rem 2rem;
}

.Signup__header {
  margin: 1.5rem 0;
}

.Signup__submit-button-div {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 7rem 0 0;
  justify-content: center;
}

.Signup__submit-button {
  width: 8.5rem;
  margin: 20px auto;
  border-radius: 4px;
  background-color: rgb(0,204,0);
  color: white;
  border-color: rgb(0,204,0);
  transition-duration: 0.4s;
}

.Signup__submit-button:hover {
  background-color: rgb(0,204,0);
}

.Signup__or {
  margin: .75rem 0;
}

.Signup__login-link {
  width: 7.5rem;
  border: none;
  background: none;
  font-weight: bolder;
  color: rgb(0,204,0);
  cursor: pointer;
  font-size: 1.8rem;
  transition-duration: 0.5ms;
}

/* .Signup__login-link:hover {
  cursor: pointer;
  background: linear-gradient(to right, #30CFD0 0%, #330867 100%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
} */

.lp_input {
  border: none;
  border-bottom: 2px solid black;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.signUpKatia {
position: initial;
margin: 0 auto;
font-weight: bold;
font-size: 40px;
border: none;
background: none;

background: linear-gradient(to right, #30CFD0 0%, #330867 100%);
background-clip: text;
-webkit-background-clip: text;
-webkit-text-fill-color: transparent;
}

.logoContainer {
  text-align: center;
  margin-top: 120px;
}

.lp_input-div {
  margin-top: 20px;
}