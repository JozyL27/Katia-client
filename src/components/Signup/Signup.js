import React, { Component } from 'react'
import {Link} from 'react-router-dom'
import TokenService from '../../services/token-service'
import AuthService from '../../services/auth-api-service'
import './Signup.css'

export default class Signup extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {}
  }

  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { email, display_name, password, confirm_password } = ev.target
    const user = {
      email: email.value,
      display_name: display_name.value,
      password: password.value
    }

    if(password.value === confirm_password.value) {
    AuthService.registerUser(user)
      .then(() => {
        AuthService.postLogin({
          email: email.value,
          password: password.value
        })
          .then(res => {
            email.value = ''
            display_name.value = ''
            password.value = ''
            TokenService.saveAuthToken(res.authToken)
            this.props.onRegistrationSuccess(user.user_name)
          })
          .catch(error => this.setState({error: error.message}))
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
    } else if(password.value !== confirm_password.value) {
      this.setState({ error: 'passwords did not match' })
    }
  }

  render() {
    const { error } = this.state
    return (
      <>
      <div className='logoContainer'>
      <Link to='/' className='lp_lp-link signUpKatia'>Katia</Link>
      </div>
      <main id='lp_main-flex'>
        <section className='Signup__container'>
          <h2 className='Signup__header'>Signup</h2>
          <form
            className='Signup__form'
            onSubmit={this.handleSubmit}
          >
            <div role='alert'>
              {error && <p className='Signup__error'>{error}</p>}
            </div>
            <div className='Signup__email-div lp_input-div'>
              <label htmlFor='Signup__email'>
                Email
              </label>
              <input
                name='email'
                type='email'
                required
                id='Signup__email'
                className='lp_input'>
              </input>
            </div>
            <div className='Signup__display_name-div lp_input-div'>
              <label htmlFor='Signup__display_name'>
                Display Name
              </label>
              <input
                name='display_name'
                type='text'
                required
                id='Signup__display_name'
                className='lp_input'>
              </input>
            </div>
            <div className='Signup__password-div lp_input-div'>
              <label htmlFor='Signup__password'>
                Password 
              </label>
              <input
                name='password'
                type='password'
                required
                id='Signup__password'
                className='lp_input'>
              </input>
            </div>
            <div className='Signup__confirm_password-div lp_input-div'>
              <label htmlFor='Signup__confirm_password'>
                Confirm Password 
              </label>
              <input
                name='confirm_password'
                type='password'
                required
                id='Signup__confirm_password'
                className='lp_input'>
              </input>
            </div>
            <button className='Signup__submit-button lp_button' type='submit'>
                Submit
              </button>
            <div className='Signup__submit-button-div'>
              {/* <button className='Signup__submit-button lp_button' type='submit'>
                Submit
              </button> */}
              <p className='Signup__or'>Already have an account? </p>
              <Link className='Signup__login-link' to='/login'>Login</Link>
            </div>
          </form>
        </section>
      </main>
      </>
    )
  }
}

