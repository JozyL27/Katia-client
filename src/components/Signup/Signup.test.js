import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Signup from './Signup'

describe('<Signup />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Signup />
      </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})