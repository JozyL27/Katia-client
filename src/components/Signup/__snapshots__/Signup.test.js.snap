// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders the UI as expected 1`] = `
<main
  id="Signup__main-flex"
>
  <section
    className="Signup__container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Project Manhattan
    </a>
    <h2>
      Signup
    </h2>
    <form
      className="Signup__form"
      onSubmit={[Function]}
    >
      <div
        role="alert"
      />
      <div
        className="Signup__user_name-div"
      >
        <label
          htmlFor="Signup__user_name"
        >
          User name
        </label>
        <input
          id="Signup__user_name"
          name="user_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="Signup__display_name-div"
      >
        <label
          htmlFor="Signup__display_name"
        >
          Display Name
        </label>
        <input
          id="Signup__display_name"
          name="display_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="Signup__password-div"
      >
        <label
          htmlFor="Signup__password"
        >
          Password
        </label>
        <input
          id="Signup__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="Signup__confirm_password-div"
      >
        <label
          htmlFor="Signup__confirm_password"
        >
          Confirm Password
        </label>
        <input
          id="Signup__confirm_password"
          name="confirm_password"
          required={true}
          type="confirm_password"
        />
      </div>
      <div
        className="Signup__submit-button-div"
      >
        <button
          className="Signup__submit-button"
          type="submit"
        >
          Signup
        </button>
        <p
          className="Signup__or"
        >
          or
        </p>
        <a
          className="Signup__login-link"
          href="/login"
          onClick={[Function]}
        >
          login
        </a>
      </div>
    </form>
  </section>
</main>
`;
